cns/api.go

```diff
@@ -382,3 +386,28 @@ type GetVMUniqueIDResponse struct {
 	Response   Response `json:"response"`
 	VMUniqueID string   `json:"vmuniqueid"`
 }
+
+// IBDevice API Contracts
+
+// AssignIBDevicesToPodRequest represents the request to assign InfiniBand devices to a pod
+// POST /ibdevices/pod
+type AssignIBDevicesToPodRequest struct {
+	PodID        string   `json:"podID"`        // podname-podnamespace format
```

Comment:
just check if the format is usable everywhere in your methods later, i.e. easy to decipher from here & then use it for checking with api-server if pod exists etc. we could also change this later if you discover it doesn't work & is efficient in other format like body/struct or separate fields

Reply:
Yeah good point, in controller-runtime, I see they do "podnamespace/podname"
<img width="487" height="90" alt="image" src="https://github.com/user-attachments/assets/5b6dc345-f5af-45ec-b71a-5d5014a101e0" />

But seeing as this will be part of the API path on the GET, and also that this API is per pod, I'd rather lead with the podname (and make it "-")

Reply:
If the accepted convention is namespaced name, you should be adapting your implementation to that convention. Please don't do your own different quirky thing just because it's convenient, that's how we (already) end up with a dozen different Pod "IDs" in different parts CNS state that are irreconcilable.
It's generally more parseable and makes more sense when lexicographically sorted if you order by least -> most specificity. Pod is more specific than namespace, so it should go after.

Reply:
Okay haha

Reply:
Hey @rbtr , what would you say is best at this point in CNS, to start moving towards a single consolidated `PodInfo` way of identifying a pod?
I see in https://github.com/Azure/azure-container-networking/blob/2997eb49b0fd1f29ea6cd81cf232f4c9b95efc8d/cns/NetworkContainerContract.go#L217

There's a `PodInfo` struct, should I use that here as well?

So
```go
PodID PodInfo `json:podID"`
```

Reply:
you can URL encode the `/` if you want your asset to be "namespace/name" and not "namespace"/"name"

Reply:
> you can URL encode the / if you want your asset to be "namespace/name" and not "namespace"/"name"

Oh yeah, this is actually not a problem anymore, because I moved the namespace/name from the url to the request body instead

Reply:
That's a good question, here's how we've designed it:
1. Pod A is created
2. MTPNC is also created for it with a finalizer
3. Devices are programed, pod runs etc.
4. Pod is deleted
5. MTPNC is only deleted, after all devices for that pod were unprogrammed (this will be enforced in our other repo)

So now I'm wondering, is the pod also kept alive, while the MTPNC for it is?
If yes:
- We're good, since the new pod can't be created until the old one is fully deleted
If no:
- Then yeah we'd have a problem haha

Reply:
is it just a finalizer or an ownerref?

Reply:
Finalizer
`finalizers.acn.azure.com/dnc-operations`

Reply:
As discussed, there is no toctou race on the pod name/namespace since the pod will be the owner of the mtpnc